spec/controllers/installation_controller_spec.rb

```diff
@@ -0,0 +1,623 @@
+
+      before do
+        sign_in(user)
+      end
+
+      context 'when there is no installation' do
+        it 'should create user and installation and redirect to step_1' do
+          expect do
+            get '/installation/create',
+                params: { user: { email: 'yukio@email.com', full_name: 'Yukio teste' },
+                          installation: { id: 1, key1: 'key1teste', key2: 'key2teste', token: 'tokenteste' } }
+          end.to change(Installation, :count).by(1)
+                                             .and change(User, :count).by(1)
+          expect(response).to redirect_to(installation_step_1_path)
+        end
+      end
+      context 'when there is an installation' do
+        context 'when installation status is completed' do
+          let!(:installation) { create(:installation, status: 'completed') }
+          it 'should not create user and installation and raise route error' do
+            first_installation.app_reload
+            expect do
+              get '/installation/create',
+                  params: { user: { email: 'yukio@email.com', full_name: 'Yukio teste' },
+                            installation: { id: 1, key1: 'key1teste', key2: 'key2teste', token: 'tokenteste' } }
+            end.to raise_error(ActionController::RoutingError)
+          end
+        end
+        context 'when installation status is in_progress' do
+          let!(:installation) { create(:installation, status: 'in_progress') }
+          it 'should create user, update installation and redirect to step_1' do
+            load "#{Rails.root}/app/controllers/application_controller.rb"
+            Rails.application.reload_routes!
+            expect do
+              get '/installation/create',
+                  params: { user: { email: 'yukio@email.com', full_name: 'Yukio teste' },
+                            installation: { id: 1, key1: 'key1teste', key2: 'key2teste', token: 'tokenteste' } }
+            end.to change(Installation, :count).by(0)
+                                               .and change(User, :count).by(1)
+            expect(response).to redirect_to(installation_step_1_path)
+            expect(last_user.full_name).to eq('Yukio teste')
+            expect(last_user.email).to eq('yukio@email.com')
+            expect(first_installation.id).to eq('1')
+            expect(first_installation.key1).to eq('key1teste')
+            expect(first_installation.key2).to eq('key2teste')
+            expect(first_installation.token).to eq('tokenteste')
+          end
+        end
+      end
+    end
+  end
+
+  describe 'setup_installation' do
+    context 'GET /accounts/1/users' do
+      context 'when there is no user and installation registered' do
+        it 'should redirect to installation new path' do
+          get '/accounts/1/users'
+          expect(response).to redirect_to(installation_new_path)
+        end
+      end
+      context 'when it is an unauthenticated user' do
```

Comment:
Esses espaçamentos ajuda a identificar que são contextos que não possuem relação

```suggestion
      end

      context 'when it is an unauthenticated user' do
```

Reply:
Boa! Adicionei essa alteração.